fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java

```diff
@@ -119,16 +124,108 @@ public int getMaxArity() {
         }
 
         return SINGLETON_TRUE;
     }
 
+    /**
+     * Determines if the value set is syntax-based.
+     * @param valueSet the value set
+     * @return true or false
+     */
+    private boolean isSyntaxBased(ValueSet valueSet) {
+        String valueSetUrl = valueSet.getUrl() != null ? valueSet.getUrl().getValue().split("\\|")[0] : null;
+        return ALL_LANG_VALUE_SET_URL.equals(valueSetUrl);
+    }
+    
+    /**
+     * Validates the element against the syntax-based value set.
+     * @param valueSet the value set
+     * @param evaluationContext the evaluation context
+     * @param elementNode the element node to validate
+     * @param strength the binding strength
+     * @return true if validation was successful, otherwise false
+     */
+    private boolean validateAgainstSyntaxBasedValueSet(ValueSet valueSet, EvaluationContext evaluationContext, FHIRPathElementNode elementNode, String strength) {
+        Element element = elementNode.element();
+        Code code = null;
+        Coding coding = null;
+        CodeableConcept codeableConcept = null;
+
+        // Determine the system/version/code or CodableConcept to validate
+        if (element.is(Code.class)) {
+            code = element.as(Code.class);
+        }
+        else if (element.is(Coding.class)) {
+            coding = element.as(Coding.class);
+        }
+        else if (element.is(CodeableConcept.class)) {
+            codeableConcept = element.as(CodeableConcept.class);
+        }
+        else if (element.is(Quantity.class)) {
+            Quantity quantity = element.as(Quantity.class);
+            coding = Coding.builder().system(quantity.getSystem()).code(quantity.getCode()).build();
+        }
+        else {
+            code = element.is(FHIR_STRING) ? Code.of(element.as(FHIR_STRING).getValue()) : Code.of(element.as(Uri.class).getValue());
+        }
+        
+        return validateCodeAgainstSyntaxBasedValuedSet(valueSet, code, coding, codeableConcept, evaluationContext, elementNode, strength);
+    }
+    
+    /**
+     * Validates the Code, Coding, CodableConcept against the syntax based value set.
+     * Only one of Code, Coding, or CodeableConept should be passed in.
+     * Issues will be added to evaluation context if validation is not successful.
+     * @param valueSet the value set
+     * @param code the code to validate
+     * @param code the coding to validate
+     * @param codeableConcept the codeable concept to validate
+     * @param evaluationContext the evaluation context
+     * @param elementNode the element node to validate
+     * @param strength the binding strength
+     * @return true if validation was successful, otherwise false
+     */
+    private boolean validateCodeAgainstSyntaxBasedValuedSet(ValueSet valueSet, Code code, Coding coding, CodeableConcept codeableConcept, EvaluationContext evaluationContext, FHIRPathElementNode elementNode, String strength) {
```

Comment:
maybe just combine it with the method above and take an Element?
then you just need one if/else block, right?  maybe a "horse apiece" though.

Reply:
Yeah, I went back and forth on that.  My goal was to minimize the number of lines of code to be added for any future syntax-based value sets (e.g. UCUM), so I thought this ended up looking cleaner.